### PR TITLE
Add more options for reindex command

### PR DESCRIPTION
**Pull request by ymao2**
## Description
Added extra options for cases-reindex command

## Self-review checklist

* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [X] (3) Tests passing
* [X] (4) Branch ready to be merged (not work in progress)
* [X] (5) No superfluous changes in diff
* [X] (6) No TODO's without new ticket numbers
* [X] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`

### Related JIRA tickets

https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&modal=detail&selectedIssue=CT-2900

### Manual testing instructions
run command with following examples 
./cts cases reindex
./cts cases reindex --non-indexed --size=500

### CODE REVIEW

File: lib/cts/cases/cli.rb [hunk 1]
```diff
@@ -270,6 +270,10 @@ def show(*args)
       end
     end
 
+    option :non_indexed, type: :boolean,
+           desc: 'Narrow down the scope of reindex to the cases which have not been indexed yet.'
+    option :size, aliases: 's', type: :numeric, default: 5000,
+            desc: 'Number of cases to for indexes. [5000]'
```
Comment (Polsonby) [inline lib/cts/cases/cli.rb:276]: I think there's a word missing, not sure what - limit or something? `Number of cases to for `

File: lib/cts/cases/reindex.rb [hunk 2]
```diff
@@ -11,8 +11,26 @@ def initialize(logger, options = {})
...
       logger.info "\n#{@count} cases reindexed"
     end
+
+    private 
+
+    def process_indexes
+      if @options[:non_indexed]
+        size = @options.fetch(:size)
+        logger.info "Start to query the cases with size #{size}."
+        cases = Case::Base.where("document_tsvector is NULL").limit(size)
```
Comment (Polsonby) [inline lib/cts/cases/reindex.rb:24]: So the idea is to pass the `reindex` task two options or none, is that right? Just wondering what happens if the `--size` option is omitted or you pass a string to it?

Should the method exit if you don't pass the size to it?

**Review by Polsonby — COMMENTED** [review]
I think this is fantastic work, @ymao2 :) Couple of questions - the only real concern I have is where to document this - I can't see it outputting the description for these options in the ./cts help output so wondering if we could add something there

File: lib/cts/cases/reindex.rb (hunk 2 above)
Comment (Polsonby) [reply lib/cts/cases/reindex.rb:24]: Interesting contrast between `@options[:non_indexed]` and `@options.fetch(:size)` - wondering from interest why use both syntaxes? They seem to behave slightly differently in edge cases e.g.  https://gist.github.com/stevenharman/5664318

Comment (ymao2) [reply lib/cts/cases/reindex.rb:24]: hmm,  didn't think about it too much, somehow I am sort of thinking the "fetch" similar as the "get" in python dict operator which will return nil even the key is not included, and [key] will raise a exception if key is not included, but after I checked the link, it is not what I thought,  and for this case,  2 doesn't make different, in that case let me change to [key]

File: lib/cts/cases/cli.rb (hunk 1 above)
Comment (ymao2) [reply lib/cts/cases/cli.rb]: changed the wording

**Review by Polsonby — APPROVED** [review]
Fab - love it! :)

**Pull request merged**